x/evm/keeper/fees_test.go

```diff
@@ -444,99 +473,133 @@ func (suite *KeeperTestSuite) TestVerifyFeeAndDeductTxCostsFromUserBalance() {
-					suite.Require().Equal(
-						fees,
-						sdk.NewCoins(
-							sdk.NewCoin(evmtypes.DefaultEVMDenom, sdkmath.NewIntFromBigInt(txData.EffectiveFee(baseFeeRes.BaseFee.BigInt()))),
-						),
-						"valid test %d failed, fee value is wrong  - '%s'", i, tc.name,
-					)
-					suite.Require().Equal(int64(0), priority)
+				fees, err := keeper.VerifyFee(txData, evmtypes.DefaultEVMDenom, baseFee, false, false, suite.ctx.IsCheckTx())
+				if tc.expectPassVerify {
+					suite.Require().NoError(err, "valid test %d failed - '%s'", i, tc.name)
+					if tc.enableFeemarket {
+						baseFeeRes, err := suite.app.EvmKeeper.BaseFee(suite.ctx, &evmtypes.QueryBaseFeeRequest{})
+						suite.Require().NoError(err)
+						suite.Require().Equal(
+							fees,
+							sdk.NewCoins(
+								sdk.NewCoin(evmtypes.DefaultEVMDenom, sdkmath.NewIntFromBigInt(txData.EffectiveFee(baseFeeRes.BaseFee.BigInt()))),
+							),
+							"valid test %d failed, fee value is wrong  - '%s'", i, tc.name,
+						)
+						if setup.decimals == evmtypes.Denom6Dec {
+							// priority will be > 0 due to scaling up the gasTip
+							// and feeCap values and keeping the same DefaultPriorityReduction. Considering priority has no effect ATM
+							suite.Require().Greater(priority, int64(0))
+						} else {
+							suite.Require().Equal(int64(0), priority)
+						}
+					} else {
+						suite.Require().Equal(
+							fees,
+							sdk.NewCoins(
+								sdk.NewCoin(evmtypes.DefaultEVMDenom, sdkmath.NewIntFromBigInt(gasPrice).Mul(sdkmath.NewIntFromUint64(tc.gasLimit))),
+							),
+							"valid test %d failed, fee value is wrong  - '%s'", i, tc.name,
+						)
+					}
 				} else {
-					suite.Require().Equal(
-						fees,
-						sdk.NewCoins(
-							sdk.NewCoin(evmtypes.DefaultEVMDenom, tc.gasPrice.Mul(sdkmath.NewIntFromUint64(tc.gasLimit))),
-						),
-						"valid test %d failed, fee value is wrong  - '%s'", i, tc.name,
-					)
+					suite.Require().Error(err, "invalid test %d passed - '%s'", i, tc.name)
+					suite.Require().Nil(fees, "invalid test %d passed. fees value must be nil - '%s'", i, tc.name)
 				}
-			} else {
-				suite.Require().Error(err, "invalid test %d passed - '%s'", i, tc.name)
-				suite.Require().Nil(fees, "invalid test %d passed. fees value must be nil - '%s'", i, tc.name)
-			}
 
-			err = suite.app.EvmKeeper.DeductTxCostsFromUserBalance(suite.ctx, fees, common.HexToAddress(tx.From))
-			if tc.expectPassDeduct {
-				suite.Require().NoError(err, "valid test %d failed - '%s'", i, tc.name)
-			} else {
-				suite.Require().Error(err, "invalid test %d passed - '%s'", i, tc.name)
-			}
-		})
+				err = suite.app.EvmKeeper.DeductTxCostsFromUserBalance(suite.ctx, fees, common.HexToAddress(tx.From))
+				if tc.expectPassDeduct {
+					suite.Require().NoError(err, "valid test %d failed - '%s'", i, tc.name)
+				} else {
+					suite.Require().Error(err, "invalid test %d passed - '%s'", i, tc.name)
+				}
+			})
+		}
 	}
 	suite.enableFeemarket = false // reset flag
```

Comment:
this should not be needed anymore since it is reset before every setup:

```go
suite.enableFeemarket = tc.enableFeemarket
suite.denomDecimals = setup.decimals
suite.SetupTest()
```